### PR TITLE
8288763: Pack200 extraction failure with invalid size

### PR DESCRIPTION
**Pull request by backwaterred**
Backports this change to jdk8 to fix the pack200 compressed size issue here as well.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8288763](https://bugs.openjdk.org/browse/JDK-8288763): Pack200 extraction failure with invalid size

### Reviewers
 * [Paul Hohensee](https://openjdk.org/census#phh) (@phohensee - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk8u-dev pull/86/head:pull/86` \
`$ git checkout pull/86`

Update a local copy of the PR: \
`$ git checkout pull/86` \
`$ git pull https://git.openjdk.org/jdk8u-dev pull/86/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 86`

View PR using the GUI difftool: \
`$ git pr show -t 86`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk8u-dev/pull/86.diff">https://git.openjdk.org/jdk8u-dev/pull/86.diff</a>

</details>

### CODE REVIEW

**Comment by backwaterred** [comment]
The bots were confused because the file path was changed. This backport is squeaky clean.

/label add clean

**Comment by backwaterred** [comment]
You win this round bots! I will wait for a review.

**Review by phohensee — APPROVED** [review]
/clean

A clean backport from 11u, lgtm.

**Comment by backwaterred** [comment]
Hi @jerboaa, I am reaching out because I haven't received a fix-yes or a fix-no response to the backport request on JBS. Could you please take a look when you have a moment?

**Comment by backwaterred** [comment]
Thanks all!

/integrate

**Comment by phohensee** [comment]
/sponsor

Got the approval.

**Pull request closed without merging**
